step_configure_hw.go

```diff
@@ -37,12 +51,12 @@ func (s *StepConfigureHW) Run(state multistep.StateBag) multistep.StepAction {
 			return multistep.ActionHalt
 		}
 
-		confNotEmpty = true
 		confSpec.MemoryMB = int64(ram)
+		parametersFlag.MemoryMBPtr = &(confSpec.MemoryMB)
 	}
 
 	ui := state.Get("ui").(packer.Ui)
-	if confNotEmpty {
+	if !parametersFlag.isEmpty() {
```

Comment:
should we handle all this stuff manually?
Can it be converted to something like `if config ==  Config{}`?
http://stackoverflow.com/a/28447372/576361